### PR TITLE
Part of bug fix package 8, DFL-3382 and DFL-3365

### PR DESCRIPTION
**Pull request by hzr**

### CODE REVIEW

**Comment by hzr** [comment]
Ping @chriskr @danfooo

File: src/style/css_shorthand_resolver.js [hunk 1]
```diff
@@ -279,6 +279,77 @@ CssShorthandResolver.shorthands = (function() {
...
+          if (declarations["animation-direction"][i].value !=
+              get_initial_value("animation-direction"))
+            sub_template.push(" ", declarations["animation-direction"][i]);
+
+          if (declarations["animation-fill-mode"][i].value !=
+              get_initial_value("animation-fill-mode"))
+            sub_template.push(" ", declarations["animation-fill-mode"][i]);
+
+          // There's always an extra space at the beginning, remove it here
+          sub_template.splice(0, 1);
```
Comment (chriskr) [inline src/style/css_shorthand_resolver.js]: Or just sub_template.shift()

File: src/style/css_shorthand_resolver.js [hunk 2]
```diff
@@ -769,6 +840,33 @@ CssShorthandResolver.shorthands = (function() {
...
+        for (var i = 0; i < len; i++)
+        {
+          template.push(
+            get_tokens(declarations["transition-property"][i]), " ",
+            get_tokens(declarations["transition-duration"][i]), " ",
+            get_tokens(declarations["transition-timing-function"][i]), " ",
+            get_tokens(declarations["transition-delay"][i]), ", "
+          );
+        }
+        template.splice(-1, 1); // Remove the last ','
```
Comment (chriskr) [inline src/style/css_shorthand_resolver.js]: Or just template.pop()

**Comment by chriskr** [comment]
Yes, noted that about the repeated code. Looks good.

**Comment by danfooo** [comment]
Looks good.

**Pull request merged**